### PR TITLE
openthread: align libraries for thread 1.4 and nrf54l15

### PR DESCRIPTION
**Pull request by maciejbaczmanski**
- allow building from libraries for nrf54l15 target
- allow building from libraries for Thread 1.3 and 1.4
- remove experimental mark for TCP in OT and set it to y for T1.4
- pull nrfxlib mbedtls path fixes

### CODE REVIEW

File: subsys/net/openthread/Kconfig.defconfig [hunk 1]
```diff
@@ -251,6 +251,15 @@ endif # OPENTHREAD_CSL_RECEIVER
 
 endif # !OPENTHREAD_THREAD_VERSION_1_1
 
+if OPENTHREAD_THREAD_VERSION_1_4
+
+# Thread 1.4 dependencies
+config OPENTHREAD_TCP_ENABLE
+	bool
```
Comment (ArekBalysNordic) [inline subsys/net/openthread/Kconfig.defconfig:258]:
```suggestion
```

Comment (ArekBalysNordic) [reply subsys/net/openthread/Kconfig.defconfig:258]: Or ignore it since all others are done in the same way.

Comment (nordicjm) [reply subsys/net/openthread/Kconfig.defconfig:258]: Do not ignore, remove, do not re-declare existing types for symbols that already exist

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: subsys/net/openthread/Kconfig.defconfig [hunk 2]
```diff
@@ -251,6 +251,15 @@ endif # OPENTHREAD_CSL_RECEIVER
...
+	default y
+
+endif # OPENTHREAD_THREAD_VERSION_1_4
+
 if OPENTHREAD_COPROCESSOR_RCP
 
 config MBEDTLS_PSA_KEY_SLOT_COUNT
```
Comment (nordicjm) [inline subsys/net/openthread/Kconfig.defconfig:265]: No to the below too, remove types from all symbols that already exist

Comment (maciejbaczmanski) [reply subsys/net/openthread/Kconfig.defconfig:264]: This is not in the scope of this PR. I can create a follow-up PR removing types from symbols that already exist.

File: subsys/net/openthread/Kconfig.defconfig (hunk 1 above)
Comment (maciejbaczmanski) [reply subsys/net/openthread/Kconfig.defconfig]: ok, removed.

**Review by nordicjm — APPROVED** [review]

File: subsys/net/openthread/Kconfig.defconfig (hunk 2 above)
Comment (maciejbaczmanski) [reply subsys/net/openthread/Kconfig.defconfig:264]: addressed in https://github.com/nrfconnect/sdk-nrf/pull/17707

**Pull request merged**